caffe2/serialize/istream_adapter.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <istream>
+
+#include <caffe2/serialize/read_adapter_interface.h>
+
+namespace at {
+
+// this is a reader implemented by std::istream
+class IStreamAdapter : public ReadAdapterInterface {
+ public:
+  explicit IStreamAdapter(std::istream* istream) : istream_(istream) {}
+
+  size_t size() override {
+    auto prev_pos = istream_->tellg();
+    istream_->seekg(0, istream_->end);
+    auto result = istream_->tellg();
+    istream_->seekg(prev_pos);
+    validate("getting size");
+    return result;
+  }
+
+  size_t read(uint64_t pos, void* buf, size_t n, const char* what) override {
+    istream_->seekg(pos);
+    if (!*istream_) {
+      return 0;
+    }
+    istream_->read(static_cast<char*>(buf), n);
+    if (!*istream_) {
+      return 0;
+    }
+    return n;
+  }
+
+ private:
+  std::istream* istream_;
```

Comment:
I think let the caller to maintain the ownership is acceptable. Since this IStreamAdapter is only used for PyTorchStreamReader. I can change it to shared_ptr in a different PR, because it needs to change import_ir_module as well, which is out of the scope of this PR.

Reply:
It is idiomatic to pass borrowed references to std::ostreams. Consider how operator<< typically works. It will be impossible to use with std::cout if this is changed to an std::shared_ptr, so please do not make this change.